### DISCUSSION TITLE
The right configuration to using customManagers with regex

### DISCUSSION THREAD

**Discussion by flobb** [renovatebot/renovate#29344] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

self-hosted GitLab with ghcr.io/renovatebot/renovate:37.381.7-slim

### Please tell us more about your question or problem

I can't find the right configuration to read a Dockerfile and catch, with a regex, the version stored in a env var of a Dockerfile (see below).
In the log i can see that it read the customManagers part and find the Dockerfile using the `fileMatch` regex but doesn't found any result later. I've valided the `matchStrings` on regex101.com and it look working and it's very similar to the one given in the doc. I've tried many variation of the configuration without success.

Can you help me to spot what's wrong ?

Here the configuration:
```
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "enabledManagers": [
        "custom.regex"
    ],
    "customManagers": [
        {
            "customType": "regex",
            "fileMatch": ["(^|/|\\.)Dockerfile$"],
            "matchStrings": ["ENV DD_PHP_TRACER_VERSION (?<currentValue>.*?)\\n"],
            "matchDatasources": ["github-releases"],
            "matchDepNames": ["DataDog/dd-trace-php"]
        }
    ]
}
```

Here the dockerfile:
```
FROM xxx AS stage-common

USER root

# PHP extension conf

# Datadog PHP tracer install
ENV DD_PHP_TRACER_VERSION 0.98.1
RUN datadog-ext-install.sh ${DD_PHP_TRACER_VERSION}
```

### Logs (if relevant)

```
DEBUG: Using file match: (^|/|\.)Dockerfile$ for manager regex (repository=xxx/xxx)
DEBUG: Matched 2 file(s) for manager regex: docker/app/Dockerfile, pulumi/docker/Dockerfile (repository=xxx/xxx)
DEBUG: manager extract durations (ms) (repository=xxx/xxx)
       "managers": {"regex": 2}
DEBUG: Found 0 package file(s) (repository=xxx/xxx)
DEBUG: Manager explicitly enabled in "enabledManagers" config, but found no results. Possible config error? (repository=xxx/xxx)
       "manager": "custom.regex"
 INFO: Dependency extraction complete (repository=xxx/xxx, baseBranch=main)
       "stats": {"managers": {}, "total": {"fileCount": 0, "depCount": 0}}
```

**Comment by rarkins** [comment, marked as answer]
matchDatasources and matchDepNames are invalid values in custom managers. You need datasourceTemplate and depNameTemplate instead. We have a lot of examples of custom managers both in our docs as well as in solved discussions in this repo you can follow

**Reply by flobb** [reply]
You're right it was a bad C/C after a test and i didn't realise that i haven't reverted this part.
After using `datasourceTemplate` and `depNameTemplate`, everything work. Sorry for the inconvenance.
